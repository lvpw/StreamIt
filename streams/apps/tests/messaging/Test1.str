void->void pipeline Test1 {
    portal<Counter> p;
    add Source(p);
    add Counter to p;
    add Output;
}

void->int filter Source(portal<Counter> p) {
    int i=100;
    int j=0;
    work push 1 {
	i++;
	push(i);
	if (i==110) {
	    i = 100;
	    j++;
	    p.setVal(j) [0:0];
	}
    }
}

int->int filter Counter {
    int val = 0;
    
    work push 2 pop 1 {
	push(pop());
	push(val);
    }

    handler setVal(int v) {
	val = v;
    }
}

int->void filter Output {
  work pop 1 {
    println(pop());
  }
}

/* OUTPUT
cagfarm-44 % strc -library Test1.str | & more
101
0
102
0
103
0
104
0
105
0
106
0
107
0
108
0
109
0
Delivering message setVal to Counter
110
1
101
1
102
1
103
1
104
1
105
1
106
1
107
1
108
1
109
1
Delivering message setVal to Counter
110
2
101
2
102
2
103
2
104
2
105
2
106
2
107
2
108
2
109
2
Delivering message setVal to Counter
110
3
101
*/
