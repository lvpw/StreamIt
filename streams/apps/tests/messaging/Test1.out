101
0
102
0
103
0
104
0
105
0
106
0
107
0
108
0
109
0
110
1
101
1
102
1
103
1
104
1
105
1
106
1
107
1
108
1
109
1
110
2
101
2
102
2
103
2
104
2
105
2
106
2
107
2
108
2
109
2
110
3
101
3
102
3
103
3
104
3
105
3
106
3
107
3
108
3
109
3
110
4
101
4
102
4
103
4
104
4
105
4
106
4
107
4
108
4
109
4
110
5
101
5
102
5
103
5
104
5
105
5
106
5
107
5
108
5
109
5
110
6
101
6
102
6
103
6
104
6
105
6
106
6
107
6
108
6
109
6
110
7
101
7
102
7
103
7
104
7
105
7
106
7
107
7
108
7
109
7
110
8
101
8
102
8
103
8
104
8
105
8
106
8
107
8
108
8
109
8
110
9
101
9
102
9
103
9
104
9
105
9
106
9
107
9
108
9
109
9
110
10
